### PR TITLE
Friendships v1

### PR DESCRIPTION
**Pull request by meronogbai**
- Created a friendship model with associations and all requested features for friendships
- Users can send an invitiation for friendhsip
- Users can see pending friendships
- Users can accept friendship invitations
- Wrote some unit and integration tests

### CODE REVIEW

File: app/views/users/pending_friends.html.erb [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+    <ul class="users-list">
+      <% @pending_friends.each do |friend| %>
+        <li>Name: <%= friend.name %>
+          <span class="profile-link">
+            <%= link_to 'See Profile',  user_path(friend), class: 'profile-link' %>
+          </span>    
+        </li>
+      <% end %>
+    </ul>
+  <% end %>
```
Comment (kobiyoyo) [inline app/views/users/pending_friends.html.erb]: - [x] View-related logic should be encapsulated in View helpers (app/helpers/*_helper.rb)

File: app/views/users/_invite.html.erb [hunk 2]
```diff
@@ -0,0 +1,5 @@
+<% if current_user.friend_requests.include?(user) %>
+  <%= link_to "Accept friendship", confirm_friend_path(user), class: 'profile-link' %>
+<% elsif !current_user.pending_friends.include?(user) && current_user!=user && !current_user.friend?(user) %>
+  <%= link_to "Invite to friendship", invite_friend_path(user), class: 'profile-link' %>
+<% end %>
```
Comment (kobiyoyo) [inline app/views/users/_invite.html.erb]: - [x] View-related logic should be encapsulated in View helpers (app/helpers/*_helper.rb)

File: app/views/layouts/application.html.erb [hunk 3]
```diff
@@ -17,8 +17,14 @@
...
+          <div class="menu-item">
+            <%= menu_link_to 'Pending friends', pending_friends_path %>
+          </div>
+        <% end %>
       </div>
       <% if current_user %>
+        <%= current_user.name %>
         <%= link_to 'Sign out', destroy_user_session_path, method: :delete %>
       <% else %>
         <%= link_to 'Sign in', user_session_path %>
```
Comment (kobiyoyo) [inline app/views/layouts/application.html.erb]: - [x] View-related logic should be encapsulated in View helpers (app/helpers/*_helper.rb)

**Review by kobiyoyo — CHANGES_REQUESTED** [review]
## Hi Team    
#### Great Job so far, everything works fine, but there are a few more things required. 
### General 
- [x] Let's implement linter checks for this project.
- [x] Posts timeline includes posts written by logged-in user and their friends ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/final_project/milestone5/friends_timeline.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49736080/))
- [x] User can reject a friendship request ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/final_project/milestone5/accept_reject_friendship.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49736080/))
- [x] Your test suite is failing, Lets fix that
### Make changes and submit another code review request.
*Happy coding* 
(Any questions Feel free to contact me on slack @kobiyoyo)

File: app/views/users/pending_friends.html.erb (hunk 1 above)
Comment (meronogbai) [reply app/views/users/pending_friends.html.erb]: Done

File: app/views/users/_invite.html.erb (hunk 2 above)
Comment (meronogbai) [reply app/views/users/_invite.html.erb]: Done.

File: app/views/layouts/application.html.erb (hunk 3 above)
Comment (meronogbai) [reply app/views/layouts/application.html.erb]: Done

**Review by teekaytech — CHANGES_REQUESTED** [review]
## Status: Changes Required :warning: :vertical_traffic_light: 

### Hello team :wave:

__You've made laudable progress in this milestone. The highlighted changes by the previous reviewer have been taken care of, except for one thing. Please check my comments below and fix them as appropriate.__

**[ REQUIRED CHANGES ]**

- [x] **Posts timeline should include only posts written by logged-in user and their friends ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/final_project/milestone5/friends_timeline.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49736080/))**
- Right now in your app, a user can see posts of other users even when they are not friends. Your should re-factor your project logic so that it only shows the posts written by a logged-in user and their friends alone.

- [x] **There is a button to invite a user to friendship on the users' list page and on a single-user page ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/final_project/milestone5/user_can_see_invite_buttton.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49736080/))**
- There seems to be a bug somewhere. For instance, when I click on `All Users` to `See profile` of a user, I'm having this error page:





* * *
**Please make the required changes and submit another review request when you are done.**

* * *

**Taofeek Olalere
[Twitter](https://twitter.com/ola_lere) | [Github](https://github.com/teekaytech) | @taofeekolalere on Slack**
__Happy Coding :smiley:__

**Review by kobiyoyo — APPROVED** [review]
# Project Approved :thumbsup:
## Hello Team  ,  you have done a brilliant job with this project. :clap:
#### You can merge !!!

**Pull request merged**